calico_node/startup/startup.go

```diff
@@ -781,83 +855,154 @@ func ensureDefaultConfig(cfg *api.CalicoAPIConfig, c *client.Client, node *api.N
 	// for containers to speak to services running on the host (e.g. k8s
 	// pods speaking to k8s api-server, and mesos tasks registering with agent
 	// on startup).  Note: KDD does not yet support per-node felix config.
-	if cfg.Spec.DatastoreType != api.Kubernetes {
-		if err := ensureNodeFelixConfig(c, node.Metadata.Name, "DefaultEndpointToHostAction", "RETURN"); err != nil {
-			return err
+	if cfg.Spec.DatastoreType != apiconfig.Kubernetes {
+		felixNodeCfg, err := c.FelixConfigurations().Get(ctx, fmt.Sprintf("%s%s", FELIX_NODE_CONFIG_NAME_PREFIX, node.Name), options.GetOptions{})
+		if err != nil {
+			// Create the default config if it doesn't already exist.
+			if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
+				newFelixNodeCfg := api.NewFelixConfiguration()
```

Comment:
Not sure, I've ported it as it is, but good question. @caseydavenport might know?

Reply:
I suspect it's because of hybrid-cluster type use-cases, where `calico/node` might set a value that doesn't make sense for say a Felix running on OpenStack or something like that.

Not really sure though, just guessing.

Reply:
I guess I'm trying to avoid needing per-node configs if at all possible.  In this case perhaps we should just add this to the config file that felix will load if this is a hard requirement.

Reply:
@fasaxc @neiljerram @matthewdupre do you guys know if we need to set `DefaultEndpointToHostAction` to anything other than `RETURN` by default for any of the orchestrators? Can it default to that in Felix? And if other orchestrators need something else then we can set it to other value as part of their setup/manifest

Reply:
I think the Felix default is currently DROP. If we changed that to RETURN, we'd need (if being sufficiently conscientious) OpenStack deployment methods to install different Felix config depending on which version of the Felix code they were installing. Which would not be the end of the world, but another little detail to have to keep track of.

Reply:
Just confirmed that the current Felix default is DROP.

Reply:
On the other hand, I don't _think_ there's any fundamental reason why the default with OpenStack has to be DROP, and it's arguably weird for Calico to default for OpenStack than it does for other orchestrators.

Also, fwiw, haven't yet understood the argument for why, if a different default is needed, it has to be done with per-node config.

Reply:
*to default differently

Reply:
Thanks, @neiljerram that makes sense. 
> I don't think there's any fundamental reason why the default with OpenStack has to be DROP

Des that mean changing the Felix default to `RETURN` wouldn't break OpenStack?  

PS: we set this to `ACCEPT` for [all the k8s manifests](https://docs.projectcalico.org/master/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml), so setting this to `RETURN` here doesn't seem to be doing much for k8s.

TL;DR as I understand this:
- Felix defaults to `DROP` (which may or may not be needed for OpenStack)
- Calico-node startup sets it to `RETURN`
- All the k8s manifests then set it to `ACCEPT`

If we're unsure about all the orchestrators then we can leave this as it is for now.

WDYGT? @caseydavenport @robbrockbank @fasaxc @neiljerram

Reply:
I'll leave this to others to decide - I'm happy either way, but would err on the side of "leave it as it is" since it feels safer and simpler.

Reply:
@gunjan5 I believe it won't fundamentally break OpenStack.  It will hurt release-to-release compatibility of how a Calico/OpenStack deployment behaves - or else add a lot of complexity to what we need to do in all our install methods to ensure that we have consistent behavior.

On balance, I favor preserving the existing behavior.

Reply:
Ok so, in general, we seem to have an agreement not to change the default for now and continue with the existing behavior. I'll keep this behavior as it is in calico-node startup then. Thanks all